cloudstack/data_source_cloudstack_template.go

```diff
@@ -0,0 +1,144 @@
+package cloudstack
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+	"log"
+	"regexp"
+	"time"
+)
+
+func dataSourceCloudstackTemplate() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudstackTemplateRead,
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
+			"templatefilter": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// Computed values
+			"template_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"account": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"displaytext": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceCloudstackTemplateRead(d *schema.ResourceData, meta interface{}) error {
+	cs := meta.(*cloudstack.CloudStackClient)
+	ltp := cloudstack.ListTemplatesParams{}
+	ltp.SetListall(true)
+	ltp.SetTemplatefilter(d.Get("templatefilter").(string))
+	csTemplates, err := cs.Template.ListTemplates(&ltp)
+	if err != nil {
+		log.Printf("[ERROR] Failed to list templates: %s", err)
```

Comment:
Is logging the error enough here? I assume we should also return the error here as well, right?